src/types.js

```diff
@@ -203,6 +203,15 @@ dpsl.AvailableRoutinesList;
  */
 dpsl.RoutineStatus;
 
+/**
+ * Params object of dpsl.diagnostics.power.runAcPowerRoutine()
+ * @typedef {{
+ *   expectedStatus: !string,
+ *   expectedPowerType?: string
```

Comment:
!string

Reply:
Are you sure? I am not an expert with this notion, but dpsl.RoutineStatus also denotes the optional `output?: string` and not `!string`.

Reply:
The way I see it, expectedPowerType? it either exists or not (optional). However, !string is saying that the value must not be null.

I couldn't find supporting resources, so feel free to drop this comment.